src/event.rs

```diff
@@ -0,0 +1,530 @@
+		amount_msat: u64,
+		/// The wrapped LDK event.
+		inner: ldk_events::Event,
+	},
+	// TODO: Implement after a corresponding LDK event is added.
+	//ChannelOpened {
+	//},
+	/// A channel has been closed.
+	ChannelClosed {
+		/// The channel_id of the channel which has been closed.
+		channel_id: [u8; 32],
+		/// The wrapped LDK event.
+		inner: ldk_events::Event,
+	},
+	// TODO: Implement on-chain events when better integrating with BDK wallet sync.
+	//OnChainPaymentSent {
+	//},
+	//OnChainPaymentReceived {
+	//}
+}
+
+impl Readable for LdkLiteEvent {
+	fn read<R: lightning::io::Read>(
+		reader: &mut R,
+	) -> Result<Self, lightning::ln::msgs::DecodeError> {
+		match Readable::read(reader)? {
+			0u8 => {
+				let payment_hash: PaymentHash = Readable::read(reader)?;
+				let inner: ldk_events::Event = MaybeReadable::read(reader)?.unwrap();
+				Ok(Self::PaymentSuccessful { payment_hash, inner })
+			}
+			1u8 => {
+				let payment_hash: PaymentHash = Readable::read(reader)?;
+				let inner: ldk_events::Event = MaybeReadable::read(reader)?.unwrap();
+				Ok(Self::PaymentFailed { payment_hash, inner })
+			}
+			2u8 => {
+				let payment_hash: PaymentHash = Readable::read(reader)?;
+				let amount_msat: u64 = Readable::read(reader)?;
+				let inner: ldk_events::Event = MaybeReadable::read(reader)?.unwrap();
+				Ok(Self::PaymentReceived { payment_hash, amount_msat, inner })
+			}
+			//3u8 => {
+			// TODO ChannelOpened
+			//}
+			4u8 => {
+				let channel_id: [u8; 32] = Readable::read(reader)?;
+				let inner: ldk_events::Event = MaybeReadable::read(reader)?.unwrap();
+				Ok(Self::ChannelClosed { channel_id, inner })
+			}
+			//5u8 => {
+			// TODO OnChainPaymentSent
+			//}
+			//6u8 => {
+			// TODO OnChainPaymentReceived
+			//}
+			_ => Err(lightning::ln::msgs::DecodeError::InvalidValue),
+		}
+	}
+}
+
+impl Writeable for LdkLiteEvent {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), lightning::io::Error> {
+		match self {
+			Self::PaymentSuccessful { payment_hash, inner } => {
+				0u8.write(writer)?;
+				payment_hash.write(writer)?;
+				inner.write(writer)?;
+				Ok(())
+			}
+			Self::PaymentFailed { payment_hash, inner } => {
+				1u8.write(writer)?;
+				payment_hash.write(writer)?;
+				inner.write(writer)?;
+				Ok(())
+			}
+			Self::PaymentReceived { payment_hash, amount_msat, inner } => {
+				2u8.write(writer)?;
+				payment_hash.write(writer)?;
+				amount_msat.write(writer)?;
+				inner.write(writer)?;
+				Ok(())
+			}
+			//Self::ChannelOpened { .. } => {
+			//TODO
+			//}
+			Self::ChannelClosed { channel_id, inner } => {
+				4u8.write(writer)?;
+				channel_id.write(writer)?;
+				inner.write(writer)?;
+				Ok(())
+			} //Self::OnChainPaymentSent { .. } => {
+			  //TODO
+			  //}
+			  //Self::OnChainPaymentReceived { .. } => {
+			  //TODO
+			  //}
+		}
+	}
+}
+
+pub(crate) struct LdkLiteEventQueue<K: KVStorePersister> {
+	queue: Mutex<EventQueueSerWrapper>,
```

Comment:
> I was able to get things to compile with the following diff:

Ah, now I get what you meant. I mean I'm happy to do it this way, but I'm not sure I gives us much besides some overhead from the object additional initializations?

> Also, do you have a `lib.rs` that can be included in this PR (or a dependent one) so that we can get the PR to build? As is, it won't build without one. This also would've helped catch the `LdkLiteChainAccess` component not being introduced yet.

Yes, unfortunately I had to split the project in a way that makes the intermediary PRs not build. But the next PR #11 is essentially the starting point putting the initial modules and logic together. It's also what I personally use to check changes and run `cargo fmt`. Everything after that won't have this issue anymore.

Reply:
> Yeah I agree it doesn't change much, but I do find it cleaner to keep the serialization wrapper scoped to serialization only.

Mh, I'm not convinced it works with either cloning the event queues or introducing yet another `Arc` with inner mutability?

Applying above patch gives me: 
```
error[E0507]: cannot move out of dereference of `std::sync::MutexGuard<'_, VecDeque<LdkLiteEvent>>`
   --> src/event.rs:187:72
    |
187 |         self.persister.persist(EVENTS_PERSISTENCE_KEY, &EventQueueSerWrapper(*locked_queue))?;
    |                                                                              ^^^^^^^^^^^^^ move occurs because value has type `VecDeque<LdkLiteEvent>`, which does not implement the `Copy` trait
```

Reply:
I think you want `EventQueueSerWrapper` to hold a reference and pass `&*locked_queue`.

Reply:
Alright, had to introduce two separate wrappers for `Readable`/`Writeable`, but now not holding them anymore with https://github.com/lightningdevkit/ldk-lite/pull/10/commits/9e3e9db532ef211913fe9f54849aabaa6195fcc5.